src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs

```diff
@@ -128,8 +127,11 @@ public class AdminController : Controller
         {
             if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageOpenIdApplications))
                 return Unauthorized();
-            
-            if (model.UpdateClientSecret && string.IsNullOrWhiteSpace(model.ClientSecret))
+
+            if (model.Type == ClientType.Public && !string.IsNullOrEmpty(model.ClientSecret))
+                ModelState.AddModelError(nameof(model.ClientSecret), "No client secret can be set for public applications.");
+
```

Comment:
Between @jersiovic 's and your code, I'd like to remind you two that we try to add curly braces on all if/else statements, even one-liners, and not add an empty line between `if` and `else`.